system/logaccess/ElasticStack/ElasticStackLogAccess.hpp

```diff
@@ -0,0 +1,120 @@
+/*##############################################################################
+    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#pragma once
+
+#include "jlog.hpp"
+#include "jptree.hpp"
+#include "jstring.hpp"
+#include "jfile.ipp"
+
+
+#ifndef ELASTICSTACKLOGACCESS_EXPORTS
+#define ELASTICSTACKLOGACCESS_API DECL_IMPORT
+#else
+#define ELASTICSTACKLOGACCESS_API DECL_EXPORT
+#endif
+
+#define COMPONENT_NAME "ES Log Access"
+
+/* undef verify definitions to avoid collision in cpr submodule */
+#ifdef verify
+    //#pragma message("UNDEFINING 'verify' - Will be redefined by cpr" )
+    #undef verify
+#endif
+
+#include <cpr/response.h>
+#include <elasticlient/client.h>
+
+using namespace elasticlient;
+
+class ELASTICSTACKLOGACCESS_API ElasticStackLogAccess : public CInterfaceOf<IRemoteLogAccess>
+{
+private:
+    const char * type = "elasticstack";
+
+    Owned<IPropertyTree> m_pluginCfg;
+
+    StringBuffer m_globalIndexSearchPattern;
```

Comment:
style: There are a few style differences from the bulk of the code (m_ as a prefix for member variables), compound identifiers that don't use camelCase (e.g. logmessageprefix, reqloglevel)
Since this is an independent chunk of code, and esp shares some similar style that is ok.  Not using camel case does reduce readability.